app/validate_txs.go

```diff
@@ -50,6 +50,10 @@ func filterStdTxs(logger log.Logger, dec sdk.TxDecoder, ctx sdk.Context, handler
 			logger.Error("decoding already checked transaction", "tx", tmbytes.HexBytes(coretypes.Tx(tx).Hash()), "error", err)
 			continue
 		}
+
+		// Set the tx size on the context before calling the AnteHandler
+		ctx = ctx.WithTxBytes(tx)
```

Comment:
[question] does this need to be version gated? In other words is it consensus breaking?

Reply:
https://github.com/celestiaorg/celestia-app/pull/3909#discussion_r1796032074

Reply:
nit: this would have been easier to follow if it was done once in `PrepareProposal` where the context actually is created